### PR TITLE
Consume latest terrajet

### PR DESCRIPTION
**Pull request by ulucinar**
Consume latest terrajet to resolve naming issues.

### CODE REVIEW

**Comment by ulucinar** [comment]
While Sergen was giving a try to the guide, we figured out that we do not consume naming changes from latest terrajet.

**Comment by ulucinar** [comment]
Sergen, can you give the guide a retry by using the following link:
https://github.com/ulucinar/provider-jet-template/tree/23bd04d53325ef466a1de16011a887bfe7437912

If there are further issues, let's try to address them all with this PR.

**Comment by sergenyalcin** [comment]
> Sergen, can you give the guide a retry by using the following link: https://github.com/ulucinar/provider-jet-template/tree/23bd04d53325ef466a1de16011a887bfe7437912
> 
> If there are further issues, let's try to address them all with this PR.

Ok I will try @ulucinar

**Review by sergenyalcin — APPROVED** [review]
I tested the guide by using this PR. Repository template create from this PR uses the latest version of terrajet. As a result of this, the crd manifests were generated with the correct names (branch.github.jet.crossplane.io, repository.github.jet.crossplane.io)

File: apis/hashicups/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (muvaf) [inline apis/hashicups/v1alpha1/zz_generated.deepcopy.go:1]: How are these added?

File: package/crds/hashicups.template.tf.crossplane.io_orders.yaml [hunk 2]
```diff
@@ -1,186 +0,0 @@
-apiVersion: apiextensions.k8s.io/v1
```
Comment (muvaf) [inline package/crds/hashicups.template.tf.crossplane.io_orders.yaml:1]: Is there a reason we didn't add the new CRD?

**Review by muvaf — APPROVED** [review]

File: apis/hashicups/v1alpha1/zz_generated.deepcopy.go (hunk 1 above)
Comment (ulucinar) [reply apis/hashicups/v1alpha1/zz_generated.deepcopy.go]: I was using go1.17. Switched to 1.16 to remove that extra build constraint.

**Pull request merged**